### PR TITLE
M2C-22619 Fix Tabs widget with Accordion

### PR DESCRIPTION
**Pull request by ptylek**
### Description (*)

Magento tabs widget was searching for all elements with [data-role=collapsible]. If accordion was inserted into one of tabs' content, then the logic failed to assign properly triggers, headers and content for each element. Therefore, Accordion triggers worked for Tabs content and the whole functionality of both widgets was broken. More information is in the issue #22819.

### Fixed Issues (if relevant)

1. magento/magento2#22819: Magento widgets bug Tabs with Accordion

### Manual testing scenarios (*)

Create CMS page and insert code below. It includes Tab widget and Accordion in one of its tabs. 

```
<div class="product data items" data-mage-init="{&quot;mage/tabs&quot;: {&quot;openedState&quot;: &quot;active&quot;, &quot;animate&quot;: {&quot;duration&quot;: 100}, &quot;active&quot;: 1, &quot;disabledState&quot;: &quot;disabled&quot;}}">
<div class="item title" data-role="collapsible"><a class="switch" href="#tab-cars" data-toggle="trigger">Cars</a></div>
<div id="tab-cars" class="item content" data-role="content">Cars content</div>
<div class="item title" data-role="collapsible"><a class="switch" href="#tab-movies" data-toggle="trigger">Movies</a></div>
<div id="tab-movies" class="item content" data-role="content">Movies content</div>
<div class="item title" data-role="collapsible"><a class="switch" href="#tab-music" data-toggle="trigger">Music</a></div>
<div id="tab-music" class="item content" data-role="content">
<div id="accordion" data-mage-init="{
        &quot;accordion&quot;:{
            &quot;active&quot;: [1,2],
            &quot;collapsible&quot;: true,
            &quot;openedState&quot;: &quot;active&quot;,
            &quot;multipleCollapsible&quot;: true
        }}">
<div data-role="collapsible">
<div data-role="trigger">
<h4>Title 1</h4>
</div>
</div>
<div data-role="content">Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s</div>
<div data-role="collapsible">
<div data-role="trigger">
<h4>Title 2</h4>
</div>
</div>
<div data-role="content">It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</div>
<div data-role="collapsible">
<div data-role="trigger">
<h4>Title 3</h4>
</div>
</div>
<div data-role="content">Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old.</div>
</div>
</div>
</div>
```

Effect as below:
magentoGIF

### Questions or comments

More information here:
[Tabs widget | Magento 2 Developer Documentation](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_tabs.html)
[Accordion widget | Magento 2 Developer Documentation](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_accordion.html)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by krzksz — CHANGES_REQUESTED** [review]
Thanks for the contribution! I checked your changes and it seems like it would be now required for all the main elements to be direct children of main container which was not the case before. Can you please try to find a solution to filter found elements so that they are not nested in content differently?

**Comment by ptylek** [comment]
@krzksz Thanks for your reply! I pushed a commit with fix. After this change, all main elements don't have to be direct children of main container. Instead there's a new filter method added which checks whether the elements are in a nested container or not. Thus, each container's elements are correctly assigner and don't trigger on each other.

File: lib/web/mage/tabs.js [hunk 1]
```diff
@@ -102,28 +102,49 @@ define([
...
+                .filter(isNotNested);
 
             if (this.triggers.length === 0) {
                 this.triggers = this.headers;
             }
             this._callCollapsible();
         },
 
+        /**
+         * @param {HTMLElement} elem
```
Comment (krzksz) [inline lib/web/mage/tabs.js]: Please adjust the JSDoc as the `index` is missing.

Comment (krzksz) [reply lib/web/mage/tabs.js]: Can we also get some description what this method does?

**Review by krzksz — CHANGES_REQUESTED** [review]

File: lib/web/mage/tabs.js (hunk 1 above)
Comment (ptylek) [reply lib/web/mage/tabs.js]: Thanks @krzksz! Index param added, description added.

Comment (krzksz) [reply lib/web/mage/tabs.js]: There's still `elem` instead of `element`.

Comment (ptylek) [reply lib/web/mage/tabs.js]: Fixed.

**Review by krzksz — APPROVED** [review]

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-6268](https://jira.corp.magento.com/browse/ENGCOM-6268) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed 

**Actual Result:**
after

**Comment by sdzhepa** [comment]
@magento run Integration Tests

**Comment by nwmagento** [comment]
@magento run Integration Tests

**Pull request merged**